python/cudf/cudf/core/index.py

```diff
@@ -1456,6 +1456,7 @@ def __repr__(self):
                 pd_preprocess.dtype._categories = (
                     preprocess.categories.to_pandas()
                 )
+                pd_preprocess.dtype._ordered = preprocess.dtype.ordered
```

Comment:
Does putting this in the `pd.Categorical` call trigger the validation that causes the slowdown too?

Reply:
Sorry, the `CategoricalIndex` call?

Reply:
unfortunately assigning ordered in either or both of those calls don't result in `ordered` reflecting correctly in repr.

Reply:
Darn, thanks for checking.

I should have a patch that will help us avoid this in pandas 3.0 once https://github.com/pandas-dev/pandas/pull/59647 gets merged